src/overlay/OverlayManagerImpl.cpp

```diff
@@ -508,9 +515,32 @@ OverlayManagerImpl::tick()
         auto res = mResolvedPeers.get();
         storePeerList(res.known, false, false);
         storePeerList(res.preferred, true, false);
-        mPeerIPTimer.expires_from_now(PEER_IP_RESOLVE_DELAY);
-        mPeerIPTimer.async_wait([this]() { this->triggerPeerResolution(); },
-                                VirtualTimer::onFailureNoop);
+        auto retries = res.retryCount;
+
+        std::chrono::seconds retryDelay;
+        // no errors -> disable retries completely from now on
+        if (res.errors == 0 || !retries)
+        {
+            retries.reset();
+        }
+        else
+        {
+            retries = std::make_optional(retries ? (*retries + 1) : 1);
```

Comment:
it is impossible for retries to be nullptr here, as it's handled by the condition above

Reply:
fixed